sssd_test_framework/roles/keycloak.py

```diff
@@ -0,0 +1,299 @@
+        super().__init__(*args, **kwargs)
+
+    def setup(self) -> None:
+        """
+        Make sure we have admin's TGT so we can run ipa cli commands.
+        """
+        super().setup()
+        self.host.kclogin()
+
+    def kcadm(self, command: str) -> SSHProcessResult:
+        """
+        Run kcadm command on the Keycloak server
+
+        :param command: kcadm command
+        :type command: str
+        """
+        kcadm = "/opt/keycloak/bin/kcadm.sh"
+        result = self.host.ssh.exec([kcadm] + command.split())
+        return result
+
+    def user(self, name: str) -> KeycloakUser:
+        """
+        Get user object.
+
+        :param name: User name.
+        :type name: str
+        :return: New user object.
+        :rtype: IPAUser
+        """
+        return KeycloakUser(self, name)
+
+    def group(self, name: str) -> KeycloakGroup:
+        """
+        Get group object.
+
+        :param name: Group name.
+        :type name: str
+        :return: New group object.
+        :rtype: IPAGroup
+        """
+        return KeycloakGroup(self, name)
+
+    def domain(self, name: str) -> KeycloakDomain:
+        """
+        Get integration domain object.
+
+        :param name: Integration Domain name.
+        :type name: str
+        :return: New Integration Domain object.
+        :rtype: KeycloakDomain
+        """
+        return KeycloakDomain(self, name)
+
+
+class KeycloakUser(object):
+    """
+    Keycloak user management.
+    """
+
+    def __init__(self, role: Keycloak, name: str) -> None:
+        """
+        :param role: Keycloak role object.
+        :type role: Keycloak
+        :param name: User name.
+        :type name: str
+        """
+        self.role = role
+        self.name = name
+        self.id: str = ""
+
+    def add(
+        self,
+        *,
+        password: str | None = 'Secret123',
+    ) -> KeycloakUser:
+        """
+        Create new Keycloak user.
+
+        Parameters that are not set are ignored.
+
+        :param password: Password, defaults to 'Secret123'
+        :type password: str | None, optional
+        :return: Self.
+        :rtype: KeycloakUser
+        """
+        # run command to add user
+        create_user = "create users -r master " \
+                      f"-s username={self.name} " \
+                      f"-s email={self.name}@ipa.test " \
+                      "-s enabled=true"
+        result = self.role.kcadm(create_user)
+
+        # set Keycloak user ID with results of command
+        self.id = result.stderr.split()[-1].strip("'")
+
+        # run command to set user password
+        set_password = "set-password -r master " \
+                       f"--username {self.name} " \
+                       f"--new-password {password}"
+        self.role.kcadm(set_password)
+
+        return self
+
+    def delete(self) -> None:
+        """
+        Delete existing Keycloak user.
+        :return: Nothing
```

Comment:
You can take this out, if it returns nothing we don't need it to say that it returns nothing.

Reply:
removing.